src/Function.cpp

```diff
@@ -77,6 +77,53 @@ static long cur_func_idx;				// Index into FuncList that we are currently workin
 static bool param_first=true;			// Flag to track output of commas
 static int builtin_functions_cnt;
 
+void
+FunctionAttribute::print_wrapper(std::ostream &out)
+{
+	out << " __attribute__((" << FuncAttrName << "))";
+}
+
+BooleanFunctionAttribute::BooleanFunctionAttribute(int attr_prob, string attr_name)
+{
+	FuncAttrProb = attr_prob;
+	FuncAttrName = attr_name;
+}
+
+void
+BooleanFunctionAttribute::output_attr(std::ostream &out)
+{
+	if(rnd_flipcoin(FuncAttrProb))
+		print_wrapper(out);
+}
+
+MultiValuedAttribute::MultiValuedAttribute(int attr_prob, string attr_name, vector<string> attr_args)
+{
+	FuncAttrProb = attr_prob;
+	FuncAttrName = attr_name;
+	attribute_values = attr_args;
+}
+
+void
+MultiValuedAttribute::output_attr(std::ostream &out)
+{
```

Comment:
In the next push, I will replace below code as - 
`FuncAttrName += attribute_values[rnd_upto(attribute_values.size())];`
'print_wrapper(out);' 
Hence, there is no need of nested `if-else`